components/nimbus/src/enrollment.rs

```diff
@@ -946,6 +948,17 @@ where
         .collect()
 }
 
+pub(crate) fn sort_experiments_by_published_date(experiments: &[Experiment]) -> Vec<Experiment> {
+    let mut experiments = experiments.to_vec();
+    experiments.sort_by(|a, b| match (a.published_date, b.published_date) {
+        (Some(a), Some(b)) => a.cmp(&b),
+        (Some(_), None) => Ordering::Greater,
+        (None, Some(_)) => Ordering::Less,
+        (_, _) => Ordering::Equal,
+    });
```

Comment:
`Option<Ord>` also implements `Ord`: 

```suggestion
    experiments.sort_by(|a, b| a.published_date.cmp(b.published_date));
```